src/player.cpp

```diff
@@ -4904,11 +4912,11 @@ dealt_damage_instance player::deal_damage(Creature* source, body_part bp, const
             add_msg_player_or_npc(m_bad, _("%s grabs you!"), _("%s grabs <npcname>!"),
                                   source->disp_name().c_str());
             if (has_grab_break_tec() && get_grab_resist() > 0 && get_dex() > get_str() ?
-                dice(get_dex(), 10) : dice(get_str(), 10) > dice(source->get_dex(), 10)) {
+                rng(0, get_dex()) : rng( 0, get_str()) > rng( 0 , 10)) {
                 add_msg_player_or_npc(m_good, _("You break the grab!"),
                                       _("<npcname> breaks the grab!"));
             } else {
-                add_effect("grabbed", 1);
+                add_effect("grabbed", 4);
```

Comment:
If I recall correctly, this would add 4 turns of grab, while you probably want something closer to 2-3 intensity 2 turn grab.